### PR TITLE
Iteration 1 Updates to Report and Controller

### LINKED ISSUE #78: Update Education GET tests
- [ ] GET by ID
- [ ] GET all

### LINKED ISSUE #80: Update Report GET test
Update test for Get by ID with Report to account for new changes in formatting

### PR DESCRIPTION
**Pull request by priyapower**
# This PR:
---

## Changes
- Finalizes updates to Report and Controller with ALL passing tests
- [X] Adds code for functionality
- [X] Adds a test
- [ ] Updates Documentation/ReadMe
- [ ] Adds dependencies
- [X] Refactors code
- [ ] Fixes a bug

## Checklist
**Please check off what you can for this PR:**
- [X] This code passes all tests on local machine
- [X] This code is DRY
- [X] This code follows OOP
- [X] This code, if updates to functionality were made, updates all corresponding files

**The "I have" section is a double check for PR specific information:**
- [X] I have tagged reviewers for this PR
- [X] I have checked what branch I am merging into (current expectation is `it1_development`)
- [X] I have added this PR to the project
- [X] I have added this PR to the correct milestone
- [ ] I have updated documenation if changes were made that affect setup/running

## Linked tickest / issues 
Closes #78 
Closes #80

### CODE REVIEW

File: api/controller/EducationController.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def destroy_education(self, id):
                         "url": f'/api/v1/education/{id}'
                     }
                 }
-            }, 204
+            }, 202
```
Comment (Arique1104) [inline api/controller/EducationController.py:127]: Glad we know now how to solve this red herring.

Comment (Arique1104) [reply api/controller/EducationController.py:127]: For those of you who don't know, this is so cool!  Basically, 204 represents a no-content error...which gramatically would seem logical to put as the error response given to a delete function.  However, under the hood, python is taking this literal 204 no-content error and providing  n o   c o n t e n t.  Changing it to a 202 response allowed our coded response to be completed.

File: api/controller/ReportController.py [hunk 2]
```diff
@@ -56,8 +56,8 @@ def get_report(self, id):
                                 "downpayment_saved": report.downpayment_savings,
                                 "downpayment_percent_saved": report.percentage_saved_based_on_principal(),
                                 "plan_style": {
-                                    "01_keanu_frugal": {
-                                        "saving_style_percentage": 0.1,
+                                    "min_savings_plan": {
+                                        "savings_style_percentage": 0.2,
```
Comment (Arique1104) [inline api/controller/ReportController.py:60]: Glad we're getting real names for these keys!

File: api/services/zip.py [hunk 3]
```diff
@@ -4,16 +4,22 @@
...
+    # city = json_response['city']
+    # state = json_response['state']
+    # return f'{city}, {state}'
 
-    response = requests.get(url)
-    json_response = json.loads(response.text)
-    city = json_response['city']
-    state = json_response['state']
-    return f'{city}, {state}'
+    # UNTIL YOU MOCK THE BEHAVIOR ABOVE
```
Comment (Arique1104) [inline api/services/zip.py:21]: We should convert this to be live.

File: app.py [hunk 4]
```diff
@@ -1,34 +1,22 @@
...
 }
 
-"""For Testing or Dev Purposes Only"""
-# # please comment out if not using
-
-# app.config['MONGODB_SETTINGS'] = {
-#     'db': os.environ['DBNAME'],
-#     'host': os.environ['HOST']
-# }
-
```
Comment (Arique1104) [inline app.py:31]: So glad you figured out how to get rid of this.

File: database/report.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 imaginative = {
-    "zipcode": 11111,
-    "credit_score": 695,
-    "salary": 55000,
+    "zipcode": 60654,
+    "credit_score": 617,
+    "salary": 3000,
```
Comment (Arique1104) [inline database/report.py:4]: These were an awesome exercise of edge-testing and sad-pathing for this code.

**Review by Arique1104 — CHANGES_REQUESTED** [review]
I'd check api/services/zip.py before we merge this to heroku

File: api/services/zip.py (hunk 3 above)
Comment (priyapower) [reply api/services/zip.py:21]: For testing - we need a MOCK
For deployment - this needs to be live, yes 😸

File: app.py (hunk 4 above)
Comment (priyapower) [reply app.py:31]: I'm actually concerned that our Pipenv run of tests is hitting production database on Atlas - we should confirm this together when we work on our push to production next

File: database/report.py (hunk 5 above)
Comment (priyapower) [reply database/report.py:4]: YES! I can't believe how long it took for me to find out that our code was breaking because we were using a "fake" zipcode!

**Review by Arique1104 — APPROVED** [review]

**Pull request merged**